scripts/src/saforcharttesting/saforcharttesting.py

```diff
@@ -214,6 +214,28 @@ def write_sa_token(namespace, token):
                 with open(token, "w") as fd:
                     fd.write(base64.b64decode(content).decode("utf-8"))
 
+def switch_project_context(namespace, token):
+    tkn = open(token).read()
+    for i in range(7):
+        out = subprocess.run(["./oc", "login", "--token", tkn, "--server", "https://api.ocpappsvc-osd.zn6c.p1.openshiftapps.com:6443"], capture_output=True)
+        stdout = out.stdout.decode("utf-8")
+        print(stdout)
+        out = subprocess.run(["./oc", "project", namespace], capture_output=True)
+        stdout = out.stdout.decode("utf-8")
+        print(stdout)
+        out = subprocess.run(["./oc", "config", "current-context"], capture_output=True)
+        stdout = out.stdout.decode("utf-8").strip()
+        print(stdout)
+        if stdout.endswith(":".join((namespace, namespace))):
+            print("current-context:", stdout)
+            return
+        time.sleep(10)
+
+    # This exit will happen if there is an infra failure
+    print("""[ERROR] There is an error creating namespace and service account.
+This is an infrastructure failure.  It is not directly related to the changes in the pull request.""")
```

Comment:
Would it help to use to tell him to retry the job?

Reply:
Yes. See this change. db4d14e505780f59c6acb74310b542caff31dfd0